src/app/blogpost/post-details/post-details.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+
+import { PostActions } from 'src/app/state/blogpost/actions';
+import { getPostById } from 'src/app/state/blogpost/blogpost.selectors';
+import { State } from 'src/app/state/blogpost/blogpost.state';
+import { Post } from '../models';
+
+@Component({
+  selector: 'bpt-post-details',
+  templateUrl: './post-details.component.html',
+  styleUrls: ['./post-details.component.scss'],
+})
+export class PostDetailsComponent implements OnInit {
+  protected post$!: Observable<Post>;
+
+  constructor(private store: Store<State>, private route: ActivatedRoute) {}
+
+  ngOnInit(): void {
+    const postId = +this.route.snapshot.params['id'];
```

Comment:
This will not refresh screen on url changes:
try to add to html:    <a [routerLink]="['/posts', post.id! + 1]"> NEXT </a>

simplest solution (but kind-of dirty because of load side-effect) could be to map id to post using pipes e.g.:
```
protected post$ = this.route.paramMap.pipe(
    filter((params) => Boolean(params.get('id'))),
    map((params) => Number(params.get('id'))),
    distinctUntilChanged(),
    switchMap((id) => {
      this.store.dispatch(
        PostActions.loadPost({ id })
      );
      return this.store.select(getPostById(id));
    }),
    shareReplay()
  );
  ```

As more advanced approach - I would recommend to split this component into 2 _smart_ and _dumb_
https://blog.angular-university.io/angular-2-smart-components-vs-presentation-components-whats-the-difference-when-to-use-each-and-why/
e.g. if we would like to show multiple post details on single page